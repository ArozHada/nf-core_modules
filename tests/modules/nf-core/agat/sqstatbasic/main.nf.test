nextflow_process {

    name "Test Process AGAT_SQSTATBASIC"
    script "modules/nf-core/agat/sqstatbasic/main.nf"
    config "tests/modules/nf-core/agat/sqstatbasic/nextflow.config"
    process "AGAT_SQSTATBASIC"
    tag "agat_sqstatbasic

    test("test_basic") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [id:'test'], file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true) ]
                """
            }
        }

        then {
            // Assert that the process ran successfully
            assert process.success

            assert snapshot(process.out).match("test_basic")
        }

    }

}
