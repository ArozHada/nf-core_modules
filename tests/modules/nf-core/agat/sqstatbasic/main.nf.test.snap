{
    "Sarscov2 Genome GFF": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.stats.txt:md5,a41a0a9cfba3ed7cc1aff28815843c78"
                    ]
                ],
                "1": [
                    "versions.yml:md5,99b5257d7ffde0cdd1a3e99e4b5e9f6a"
                ],
                "stats_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.stats.txt:md5,a41a0a9cfba3ed7cc1aff28815843c78"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,99b5257d7ffde0cdd1a3e99e4b5e9f6a"
                ]
            }
        ],
        "timestamp": "2023-04-05T18:06:21+0000"
    },
    "sarscov2_genome_gff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.stats.txt:md5,a41a0a9cfba3ed7cc1aff28815843c78"
                    ]
                ],
                "1": [
                    "versions.yml:md5,99b5257d7ffde0cdd1a3e99e4b5e9f6a"
                ],
                "stats_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.stats.txt:md5,a41a0a9cfba3ed7cc1aff28815843c78"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,99b5257d7ffde0cdd1a3e99e4b5e9f6a"
                ]
            }
        ],
        "timestamp": "2023-04-05T18:06:21+0000"
    }
}